src/service-worker.js

```diff
@@ -47,3 +47,24 @@ workbox.routing.registerRoute(
     plugins: [new workbox.expiration.Plugin({ maxEntries: 10 })],
   }),
 );
+
+self.addEventListener("activate", () => {
+  self.clients.claim();
+});
+
+self.addEventListener("fetch", (event) => {
+  if (event.request.method !== "POST") {
```

Comment:
This is headed in the right direction, but you want to be sure to only capture requests targeted at the /web-share-target fake URL. It’s probably easiest if you adapt my minimal example flow in this demo app: https://github.com/tomayac/fugu-greetings/search?q=share+target&type=.